app/src/main/java/com/example/iou/map/models/RestaurantItem.java

```diff
@@ -0,0 +1,65 @@
+package com.example.iou.map.models;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RestaurantItem {
+    private String name;
+    private String address;
+    private double rating;
+    private double latitude;
+    private double longitude;
+    private Boolean isOpen;
+
+    public RestaurantItem(JSONObject jsonObject) throws JSONException {
+        this.name = jsonObject.getString("name");
+        this.address = jsonObject.getString("vicinity");
+        this.latitude = jsonObject.getJSONObject("geometry").getJSONObject("location").getDouble("lat");
+        this.longitude = jsonObject.getJSONObject("geometry").getJSONObject("location").getDouble("lng");
+
+        try {
+            this.isOpen = jsonObject.getJSONObject("opening_hours").getBoolean("open_now");
+        }
+        catch (JSONException e) {
+            isOpen = null;
+        }
+        try {
+            this.rating = jsonObject.getDouble("rating");
+        }
+        catch (JSONException e) {
+            this.rating = -1;
+            e.printStackTrace();
+        }
```

Comment:
You can use `jsonObject.has()` to check if a field exists on the object, rather than having to wrap this code in try-catch blocks! There's also some optional versions of the get functions, e.g. `optDouble()`, that will returned the value if it exists, and a specified fallback value otherwise.

Android docs on JSONObject, for more info on the functions: https://developer.android.com/reference/org/json/JSONObject